src/movement/CommunityBasedMovement.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright 2010 Aalto University, ComNet
+ * Released under GPLv3. See LICENSE.txt for details.
+ */
+
+/**
+ * CommunityBasedMovement where the coordinates are restricted to rectangle
+ * area defined by two coordinates in the entire map.
+ * 
+ * Four parameters for configuration file:
+ * CommunityBasedMovement.communityNumber: the number of rectangle community
+ *		e.g. if you have 12 communities, you could set the value to 3,4 or 4,3
+ * CommunityBasedMovement.mapSize: the size of map should be identical to worldSize
+ * 		e.g. CommunityBasedMovement.mapSize = x,y : x,y are the maximum coordinates
+ * 			in X-Axis and Y-Axis respectively 
+ * CommunityBasedMovement.communityID = a,b,c,d: all community identity
+ * CommunityBasedMovement.probabilities_local_roaming: probabilities for p_local and 
+ * 		p_roaming
+ * 
+ * Note: To run this movement model:
+ * 1. Move the configuration file "communitybasedmovement_settings.txt" from 
+ * the directory "example_settings" to root directory
+ * 2.rename the configuration to "default_settings.txt" to replace the original
+ * one
+ * 3. recompile the code
+ * 
+ * @author Xiaoyang Zhu
+ * @email xiaoyang.zhu@hotmail.com
+ */
+package movement;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import core.Coord;
+import core.Settings;
+
+public class CommunityBasedMovement extends MovementModel {
+	/* Number of the Community: 
+	   e.g. if you have 12 communities, you could set the value to 3,4 or 4,3 */
+	public static final String COMMUNITY_NUMBER = "communityNumber";
+	/* Size of the map */
+	public static final String	MAP_SIZE = "mapSize";
+	/* Community identifier that should be identical to the Group ID in configuration file */
+	public static final String	COMMUNITY_ID = "communityID";
+	
+	public static final String COMMUNITY_BASED_MOVEMENT_MODEL_NS = "CommunityBasedMovement";
+
+
+	/* Probabilities of local or roaming epoch */
+	public static final String	PROBABILITIES_LOCAL_ROAMING = "probabilities_local_roaming";
+	
+	/* Size of the map */
+	private int		map_x_max = 100, map_y_max = 100;
+	/* Number of the Community */
+	private int 	community_x_number = 1, community_y_number = 1;
+	
+	/* Community identifier that should be identical to the Group ID in configuration file */
+	private static String[]	community_id;
```

Comment:
I'm not quite sure about this. It feels like it might be better to have explicit configuration for the home community of each group in the settings file. Or at least an option to do it explicitly.